src/providers/on-chain-quote-provider.ts

```diff
@@ -319,13 +340,42 @@ export class V3QuoteProvider implements IV3QuoteProvider {
 
   private async getQuotesManyData(
     amounts: CurrencyAmount[],
-    routes: V3Route[],
+    routes: TRoute[],
     functionName: 'quoteExactInput' | 'quoteExactOutput',
     _providerConfig?: ProviderConfig
   ): Promise<{
-    routesWithQuotes: V3RouteWithQuotes[];
+    routesWithQuotes: RouteWithQuotes<TRoute>[];
     blockNumber: BigNumber;
   }> {
+    // we cannot have both V3 and MixedRoutes in the same call
+    if (
+      routes.some((route) => route instanceof V3Route) &&
+      routes.some((route) => route instanceof MixedRoute)
+    ) {
+      throw new Error(
+        'Cannot have both V3 and MixedRoutes in the same call to on chain Quoter'
+      );
+    }
+    // cannot make an exactOutput call with mixedRouteQuotes
+    if (
+      functionName === 'quoteExactOutput' &&
+      routes.some((route) => route instanceof MixedRoute)
+    ) {
+      throw new Error('Cannot make an exactOutput call with MixedRoutes');
+    }
+
+    const isMixedRoutes = routes.every((route) => route instanceof MixedRoute);
+
+    // we only support getting mixedRouteQuotes for chains that support V2 liq
+    if (
+      isMixedRoutes &&
+      !V2_SUPPORTED.some((chainId) => chainId === this.chainId)
+    ) {
+      throw new Error(
+        `Cannot get MixedRoute quotes on ${this.chainId} because it does not support V2 liquidity`
+      );
+    }
```

Comment:
May not be necessary since we don't construct an `onChainQuoteProvider<MixedRoute>` for non V2_SUPPORTED chains but extra validation guard in case this class is exported and used as standalone